packages/playground/blueprints/src/lib/steps/install-theme.ts

```diff
@@ -61,70 +62,73 @@ export const installTheme: StepHandler<InstallThemeStep<File>> = async (
 	{ themeZipFile, options = {} },
 	progress
 ) => {
-	progress?.tracker.setCaption(
-		`Installing the ${zipNameToHumanName(themeZipFile.name)} theme`
-	);
+	const zipFileName = themeZipFile.name.split('/').pop() || 'theme.zip';
+	const zipNiceName = zipNameToHumanName(zipFileName);
+
+	progress?.tracker.setCaption(`Installing the ${zipNiceName} theme`);
 	try {
-		const activate = 'activate' in options ? options.activate : true;
+		// Extract to temporary folder so we can find theme folder name
```

Comment:
This seems to be the same logic as in `installPlugin` – it would be nice to extract it to a separate function instead of repeating it.

Reply:
Sure, I extracted the common logic between install plugin/theme steps into an `installAsset` function.

https://github.com/WordPress/wordpress-playground/pull/427/files#diff-0fe76e82e969a1cae5a9aed6d4976a180ca4946567925907b24b92d28a4c99a5

Wasn't sure about the DRY-ness or granularity. It's a single function that does a few things:

- Extract asset zip file to temporary folder
- Find extracted asset folder name and throw if not found
- Move asset folder to target path: currently either `wp-content/plugins` or `wp-content/themes`
- Clean up temporary foder

The activation step remains in `installPlugin` and `installTheme` functions respectively.